File: persistence/src/main/java/com/comssa/persistence/question/dto/major/request/RequestMakeQuestionChoiceDto.java
package com.comssa.persistence.question.dto.major.request;

import com.comssa.persistence.question.domain.common.QuestionChoice;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
public class RequestMakeQuestionChoiceDto {
	private String text;
	private boolean answerStatus;

	public static RequestMakeQuestionChoiceDto from(QuestionChoice questionChoice) {
		return RequestMakeQuestionChoiceDto.builder()
			.answerStatus(questionChoice.isAnswerStatus())
			.text(questionChoice.getContent())
			.build();
	}
}

File: persistence/src/main/java/com/comssa/persistence/question/dto/common/request/RequestMakeQuestionChoiceDto.kt
package com.comssa.persistence.question.dto.common.request

import com.comssa.persistence.question.domain.common.QuestionChoice

class RequestMakeQuestionChoiceDto(
	val text: String,
	val answerStatus: Boolean,
) {
	companion object {
		fun from(questionChoice: QuestionChoice): RequestMakeQuestionChoiceDto =
			RequestMakeQuestionChoiceDto(
				text = questionChoice.content,
				answerStatus = questionChoice.isAnswerStatus,
			)
	}
}
